### PR TITLE
Add fieldState invalid depreciation note

### LINKED ISSUE #8195: Why is Controller fieldState naming inconsistent?
In formState 👉  `isValid` - boolean - Set to true if the form doesn't have any errors.
In fieldState 👉 `invalid` - boolean - Invalid state for current input.

This threw me off a bit. Had to go to the docs to find the `invalid` return object.

Just curious if this is a purposeful difference, and if so what is the reason?

Best,
Jeremy

### PR DESCRIPTION
**Pull request by Moshyfawn**

### CODE REVIEW

**Comment by Moshyfawn** [comment]
@bluebill1049 can you check if the deprivation note shows up and the invalid state is crossed over in VSCode?

I'm from my phone atm

File: src/types/form.ts [hunk 1]
```diff
@@ -321,6 +321,12 @@ export type UseFormGetFieldState<TFieldValues extends FieldValues> = <
   name: TFieldName,
   formState?: FormState<TFieldValues>,
 ) => {
+  /**
+  * @deprecated check `fieldState.error` instead
+  * ```jsx
+  * {!!fieldState.error && <p>{fieldState.error.message}</p>}}
```
Comment (bluebill1049) [inline src/types/form.ts:327]:
```suggestion
  * {fieldState.error && <p>{fieldState.error.message}</p>}
```

Comment (bluebill1049) [reply src/types/form.ts:327]: it's safe without `!!` going to return `undefined`

**Comment by Moshyfawn** [comment]
Uh, the API report is stale. I'll be able to update it only in 5-6 hours

File: src/types/form.ts (hunk 1 above)
Comment (Moshyfawn) [reply src/types/form.ts]: Yea, makes sense!

**Comment by bluebill1049** [comment]
no worries (rush) at all @Moshyfawn 👍

**Comment by Moshyfawn** [comment]
# Results

## `getFieldState` `fieldState.invalid`

image

## `Controller` `fieldState.invalid`

image

**Comment by bluebill1049** [comment]
That's awesome, we probably can start more of those to aligned with v8.

**Review by bluebill1049 — APPROVED** [review]

**Comment by Moshyfawn** [comment]
> That's awesome, we probably can start more of those to align with v8.

Yep, I already have a draft with a couple of notes like that. It might help to prepare the user to migrate before the actual release!

**Comment by bluebill1049** [comment]
Amazing!

**Pull request merged**